### DISCUSSION TITLE
How to correctlly hanlde/setup/inject dbcontext in WPF PRISM?

### DISCUSSION THREAD

**Discussion by LudwigNel** [PrismLibrary/Prism#2727] [WPF] [answered]
Good day guys.

I haven't worked on a QPF PRISM project for a while until now. And  the memory is a little slow :)
I am cretin a reporting application in WPF and PRISM.
I am going to use the Unit of Work / Repository pattern. So this is where I had got up to.

I have my Unit of Work implementation wit the related repositories. I am just not sure how get this configured and injected correctly.

This is what the Unit of Work class looks like

  public class UnitOfWork : IUnitOfWork
    {
        private readonly ApplicationDbContext _dbContext;

        public UnitOfWork(ApplicationDbContext dbContext, IProcessStepRepository processSteps)
        {
            _dbContext = dbContext;
            ProcessSteps = processSteps;
        }

        public IProcessStepRepository ProcessSteps { get; }
        public async Task<int> CompleteAsync() => await _dbContext.SaveChangesAsync();

        public void Dispose()
        {
            _dbContext.Dispose();
        }
    }

And here is the  constructor for the Db Context

    public class ApplicationDbContext : DbContext, IApplicationDbContext
    {
        public DbSet<StepReason> StepReasons { get; set; }
        public DbSet<ProcessStep> ProcessSteps { get; set; }

        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
            : base(options)
        {

        }

Now in normal circumstances like in a web api you would call services.AddDbContext.... or something like that where you can configure the  db context with its connection string etc. in the startup class.
Is there a similar way of doing this with PRISM? or am I going about this the wrong way?

I am using a .NET 6 WPF project and class libraries in this solution.

**Comment by dansiegel** [comment, marked as answer]
the `services.AddDbContext<TContext>(...)` extension just adds a little syntax sugar around registering the DbContext. It's not really magic though and you should be able to view the source from Visual Studio. By default it will register the DbContext as a Scoped service which isn't really relevant for your WPF app... you would want to look at either registering it as a singleton or possibly a Transient if you see errors with one process trying to query the database concurrantly with another process in your app. There are a few ways you could do this but you might look at something like:
```cs
var builder = new DbContextOptionsBuilder<ApplicationDbContext>();
builder.UseSqlServer("...");
containerRegistry.RegisterInstance(builder.Options);
containerRegistry.Register<ApplicationDbContext>();
```

**Reply by LudwigNel** [reply]
Thank you very much.
You know looking at your answer here I really should have gotten this. A little embarrassed to admit I didn't.
Really appreciate your time in replying to this
